fccodegen/templates/wl/weblab_model.pyx

```diff
@@ -1,47 +1,50 @@
-
 # cython: profile=True
-## @file
-##
-## This source file was generated from CellML.
-##
-## Model: tentusscher_model_2004_epi
-##
-## Processed by pycml - CellML Tools in Python
-##     (translators: , pycml: , optimize: )
-## on Fri May 18 13:29:33 2018
-##
-## <autogenerated>
-
-import numpy as np
+#
+#
+# Web Lab model {{name}}
+#
+# Generated by {{generator}} on {{date}}
+#
+#
+#
+cimport fc.sundials.sundials as Sundials
+cimport libc.math as math
+cimport numpy as np
 
 import fc.simulations.model as Model
 import fc.utility.environment as Env
 import fc.language.values as V
-
-cimport libc.math as math
-cimport numpy as np
+from fc.sundials.solver cimport CvodeSolver
+from fc.utility.error_handling import ProtocolError
+import numpy as np
 import os
 import shutil
 import sys
 
-from fc.sundials.solver cimport CvodeSolver
-cimport fc.sundials.sundials as Sundials
-from fc.utility.error_handling import ProtocolError
 
 cdef int _EvaluateRhs(Sundials.realtype var_environment__time, Sundials.N_Vector y, Sundials.N_Vector ydot, void* user_data):
+    """
+    Main model method: evaluates the model's RHS.
+    """
+    #TODO: What's this?
     model = <object>user_data
     cdef np.ndarray[Sundials.realtype, ndim=1] parameters = <np.ndarray>model.parameters
-    # State variables
-    cdef double var_rapid_time_dependent_potassium_current_Xr1_gate__Xr1 = (<Sundials.N_VectorContent_Serial>y.content).data[0]
-    cdef double var_rapid_time_dependent_potassium_current_Xr2_gate__Xr2 = (<Sundials.N_VectorContent_Serial>y.content).data[1]
 
+    # Unpack state variables
+    {%- for state in states %}
+    cdef double {{state[1]}} = (<Sundials.N_VectorContent_Serial>y.content).data[{{state[0]}}]
+    {%- endfor %}
 
     # Mathematics
-    cdef double d_dt_rapid_time_dependent_potassium_current_Xr1_gate__Xr1 = ((1.0 / (1.0 + math.exp(( -26.0 - parameters[3]) * 0.14285714285714285))) - var_rapid_time_dependent_potassium_current_Xr1_gate__Xr1) / (1.0 * (450.0 / (1.0 + math.exp(( -45.0 - parameters[3]) * 0.10000000000000001))) * (6.0 / (1.0 + math.exp((parameters[3] + 30.0) * 0.086956521739130432)))) # per_millisecond
-    cdef double d_dt_rapid_time_dependent_potassium_current_Xr2_gate__Xr2 = ((1.0 / (1.0 + math.exp((parameters[3] + 88.0) * 0.041666666666666664))) - var_rapid_time_dependent_potassium_current_Xr2_gate__Xr2) / (1.0 * (3.0 / (1.0 + math.exp(( -60.0 - parameters[3]) * 0.050000000000000003))) * (1.1200000000000001 / (1.0 + math.exp((parameters[3] - 60.0) * 0.050000000000000003)))) # per_millisecond
+    {%- for eq in rhs_equations %}
+    cdef double {{eq}}
+    {%- endfor %}
+
+    # Pack state variable derivatives
+    {%- for state in states %}
+    (<Sundials.N_VectorContent_Serial>ydot.content).data[{{state[0]}}] = {{state[1]}}
```

Comment:
I think you want `2` not `1` :)
Another argument for `state.deriv_name` being better!

Reply:
Agreed!